src/main/java/de/tum/in/www1/artemis/service/competency/KnowledgeAreaService.java

```diff
@@ -29,32 +31,83 @@ public KnowledgeAreaService(KnowledgeAreaRepository knowledgeAreaRepository) {
      * @param knowledgeArea the knowledge area to create
      * @return the created knowledge area
      */
-    public KnowledgeArea createKnowledgeArea(KnowledgeArea knowledgeArea) {
+    public KnowledgeArea createKnowledgeArea(KnowledgeAreaDTO knowledgeArea) {
         knowledgeAreaIsValidOrElseThrow(knowledgeArea);
+        if (knowledgeArea.id() != null) {
+            throw new BadRequestException("A new knowledge cannot already have an id");
+        }
```

Comment:
We need the id when returning the data in these and other endpoints. IMO it did not make sense to create two different DTOs. I added the `@JsonProperty(access = JsonProperty.Access.READ_ONLY)` annotation to the id - this way it does not get read when recieving data from the client.

Reply:
Besides the point that I personally prefer to separate the concerns of request and response, in general, to prevent data accidentally from being able to be overwritten or seen, I think here should apply only one of these cases:

Either we think this annotation is safe enough and works, so we don't need an additional nullness check, or we have doubts that it will work, so we should use two DTOs. One reason for the widespread introduction of DTOs was to avoid having to do manual checks that someone might forget, etc.

Reply:
I actually forgot the null check, thanks 💀 I also added extra DTOs for the result.